driver/pace/driver/run.py

```diff
@@ -292,34 +394,50 @@ def from_dict(cls, kwargs: Dict[str, Any]) -> "DriverConfig":
                 "initialization_type must be one of 'baroclinic' or 'restart', "
                 f"got {initialization_type}"
             )
+
         kwargs["initialization_config"] = dacite.from_dict(
             data_class=initialization_class,
             data=kwargs.get("initialization_config", {}),
             config=dacite.Config(strict=True),
         )
+
         for derived_name in ("dt_atmos", "layout", "npx", "npy", "npz", "ntiles"):
             if derived_name in kwargs["dycore_config"]:
                 raise ValueError(
                     f"you cannot set {derived_name} directly in dycore_config, "
                     "as it is determined based on top-level configuration"
                 )
-        kwargs["dycore_config"]["layout"] = kwargs["layout"]
-        kwargs["dycore_config"]["dt_atmos"] = kwargs["dt_atmos"]
-        kwargs["dycore_config"]["npx"] = kwargs["nx_tile"] + 1
-        kwargs["dycore_config"]["npy"] = kwargs["nx_tile"] + 1
-        kwargs["dycore_config"]["npz"] = kwargs["nz"]
-        kwargs["dycore_config"]["ntiles"] = 6
-        for derived_name in ("dt_atmos", "layout", "npx", "npy", "npz"):
-            if derived_name in kwargs["physics_config"]:
-                raise ValueError(
-                    f"you cannot set {derived_name} directly in physics_config, "
-                    "as it is determined based on top-level configuration"
-                )
-        kwargs["physics_config"]["layout"] = kwargs["layout"]
-        kwargs["physics_config"]["dt_atmos"] = kwargs["dt_atmos"]
-        kwargs["physics_config"]["npx"] = kwargs["nx_tile"] + 1
-        kwargs["physics_config"]["npy"] = kwargs["nx_tile"] + 1
-        kwargs["physics_config"]["npz"] = kwargs["nz"]
+
+        kwargs["dycore_config"] = dacite.from_dict(
+            data_class=fv3core.DynamicalCoreConfig,
+            data=kwargs.get("dycore_config", {}),
+            config=dacite.Config(strict=True),
+        )
+        kwargs["physics_config"] = dacite.from_dict(
+            data_class=fv3gfs.physics.PhysicsConfig,
+            data=kwargs.get("physics_config", {}),
+            config=dacite.Config(strict=True),
+        )
```

Comment:
Is this necessary? We expect dycore_config and physics_config should always be included in the yaml.

Reply:
It is necessary because we want to use nx_tile, nz, and layout from dycore_config.